### ISSUE TITLE
Misconfigured client can cause unselected client type

### ISSUE THREAD

**Issue by linaori** [WowUp/WowUp#470]
**Describe the bug**
When the client path is not properly configured (wrong path), it can cause the "selected client" to not select anything. Even after fixing this path, this will remain unselected, making it impossible to detect addons. The select box is blending in fairly well, so hard to notice.

**To Reproduce**
When I try to enter an invalid directory, it goes back to the "My Addons" tab, so possibly another bug? When I entered an invalid directory when it's empty (`C:\Games\World of Warcraft\_retail_`), I get redirected to "My Addons", without having "Retail" selected. When I then go back to options, it selected the correct directory to my WoW directory (`C:\Games\World of Warcraft`)

1. Go to Options -> clear the retail directory, assuming this is the only client
2. Configure a directory that isn't correct
3. Get forwarded to "My Addons" and see no addons, "Selected Client" shows "World of Warcraft", and it's not clear that nothing is selected
4. After doing this once, I couldn't reproduce it again, Retail would be selected but nothing loaded, though the options auto-corrected to the correct path

**Expected behavior**
When no client is selected, I expect to have some kind of visual indicator, perhaps put the input into an error state?

Generally speaking the client selection feels a bit buggy, like it changes things underwater and doesn't reflect the changes in the input.

**Screenshots**
![image](https://user-images.githubusercontent.com/1754678/99676126-67bfd780-2a78-11eb-879b-5003f616e96d.png)
![image](https://user-images.githubusercontent.com/1754678/99676131-69899b00-2a78-11eb-98a0-b80575e15a68.png)

**Additional context**
`5` = no client
```
[2020-11-20 00:30:53.185] [info] Load-addons 5
[2020-11-20 00:30:53.211] [error] Caught error TypeError: Cannot read property 'toString' of undefined
    at t.set (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:1064964)
    at t.set lastSelectedClientType [as lastSelectedClientType] (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:1080074)
    at t.set selectedClientType [as selectedClientType] (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:1085722)
    at t.onInstalledClientsChange (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:1085342)
    at b._next (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:1085168)
    at b.__tryOrUnsub (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:674910)
    at b.next (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:674155)
    at f._next (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:673350)
    at f.next (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:673124)
    at Cm._next (file:///C:/Users/Redacted/AppData/Local/Programs/wowup/resources/app.asar/dist/main-es2015.ae288662bd3892ee270b.js:1:797763)
```

**Comment by jliddev** [comment]
Looks like a bug related to manually creating a state with 0 installed clients.

**Comment by linaori** [comment]
Probably fixed, couldn't reproduce it in 2.1.0-beta12
